diff --git a/src/include/storage/s_lock.h b/src/include/storage/s_lock.h
index 173e971..c9a6e08 100644
--- a/src/include/storage/s_lock.h
+++ b/src/include/storage/s_lock.h
@@ -686,22 +686,30 @@ typedef LONG slock_t;
 #define HAS_TEST_AND_SET
 #define TAS(lock) (InterlockedCompareExchange(lock, 1, 0))
 
+#include <intrin.h>
+
 #define SPIN_DELAY() spin_delay()
 
 /* If using Visual C++ on Win64, inline assembly is unavailable.
- * Use a _mm_pause intrinsic instead of rep nop.
+ * Use architecture-specific pause/yield intrinsics instead of rep nop.
  */
-#if defined(_WIN64)
+#if defined(_M_ARM64)
+static __forceinline void
+spin_delay(void)
+{
+	__yield();
+}
+#elif defined(_WIN64)
 static __forceinline void
 spin_delay(void)
 {
 	_mm_pause();
 }
 #else
 static __forceinline void
 spin_delay(void)
 {
 	/* See comment for gcc code. Same code, MASM syntax */
 	__asm rep nop;
 }
 #endif
