diff --git a/src/backend/libpq/be-secure-openssl.c b/src/backend/libpq/be-secure-openssl.c
index 486a66b0bf1..bdc8425d292 100644
--- a/src/backend/libpq/be-secure-openssl.c
+++ b/src/backend/libpq/be-secure-openssl.c
@@ -13,7 +13,9 @@
  *
  *-------------------------------------------------------------------------
  */
-
+#ifdef X509_NAME
+#undef X509_NAME
+#endif
 #include "postgres.h"
 
 #include <sys/stat.h>
